### ISSUE TITLE
Are MatMul supposed to srepr as Mul?

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#16430]
```python
>>> A = MatrixSymbol('A', n,n)
>>> B = MatrixSymbol('B', n,n)
>>> type(A*B)
sympy.matrices.expressions.matmul.MatMul
```
So far so good.  Now:
```
>>> srepr(A*B)
"Mul(MatrixSymbol('A', Symbol('n'), Symbol('n')), MatrixSymbol('B', Symbol('n'), Symbol('n')))"
```
Are MatMul supposed to `srepr` to Mul now after #15665?

**Comment by asmeurer** [comment]
This is because they are now subclasses. The printers fallback to subclass printers, and there isn't a srepr printer for MatMul (but there is for Mul). 

I actually forgot about the subclassing. I'm not sure if it is a good idea with the postprocessors. Let me see if anything fails if I remove it. 

Otherwise, this can be fixed by making the srepr `_print_Mul` print the class name instead of hard-coding "Mul".

**Comment by asmeurer** [comment]
There is at least one test failure, which would be fixed by https://github.com/sympy/sympy/pull/15121. 

I'll probably implement the _test_Mul fix anyway, as it's good to do that to handle subclasses. But I'll also take a second look into getting https://github.com/sympy/sympy/pull/15121 in the release.

**Comment by asmeurer** [comment]
I'm not sure about removing the subclasses. It breaks the printing, because the MatAdd printer was removed in 204081cc58e1c1a952d2f92849ed94891629a70e. Apparently before https://github.com/sympy/sympy/pull/15373 MatAdd was subclassing from AssocOp but MatMul wasn't. I think it's fine as it is. If we run into some issues where their being subclasses causes problems, we can remove them later.

**Comment by asmeurer** [comment]
It's fixed in the 1.4 release branch.
